File: MyCS50 code/DanilaDz-cs50-problems-2020-x-sentimental-cash/cash.py
# I used some examples from cash in C
from cs50 import get_float
import math


def main():
    
    # keeps prompting user for input
    while True:
        change = get_float('Change owed: ')
        cents = round(change * 100)
        if change > 0: 
            break
    
    # does all the math to get the remainder coins
    # the double slash removes the float part of the number and leaves the integer
    quarters = cents // 25
    dimes = (cents % 25) // 10
    nickels = ((cents % 25) % 10) // 5
    pennies = ((cents % 25) % 10) % 5
    
    # adds all coins together and prints them
    print(quarters + dimes + nickels + pennies)
    
main()

File: MyCS50 code/DanilaDz-cs50-problems-2020-x-cash/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>


int main()
{
    int cents;
    // keeps asking the user to input a valid number
    do
    {
        float change = get_float("Change owed: ");
        cents = round(change * 100);
    }
    while (cents <= 0);
    
    // finds out how many of each coin is needed to the total
    int quarters = cents / 25;
    int dimes = (cents % 25) / 10;
    int nickels = ((cents % 25) % 10) / 5;
    int pennies = ((cents % 25) % 10) % 5;
    // prints answer
    printf("%d\n", quarters + dimes + nickels + pennies);
}